x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -815,6 +821,143 @@ And the Investigation Guide tab should NOT be displayed
 
 TODO: add scenarios https://github.com/elastic/kibana/issues/166215
 
+### Rule upgrade workflow: simple diff algorithm
+
+#### **Scenario: Rule field doesn't have an update and has no custom value**
+
+**Automation**: 2 integration tests with mock rules + a set of unit tests for the algorithm
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And <field_name> field is not customized by the user (current version == base version)
+And <field_name> field is not updated by Elastic in this upgrade (target version == base version)
+Then for <field_name> field the diff algorithm should output the current version as the merged one without a conflict
+And <field_name> field should not be returned from the `upgrade/_review` API endpoint
+And <field_name> field should not be shown in the upgrade preview UI
+
+Examples:
+| field_name | base_version | current_version | target_version |
+| name       | "A"          | "A"             | "A"            |
+| risk_score | 1            | 1               | 1              |
+```
+
+#### **Scenario: Rule field doesn't have an update but has a custom value**
+
+**Automation**: 2 integration tests with mock rules + a set of unit tests for the algorithm
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And <field_name> field is customized by the user (current version != base version)
+And <field_name> field is not updated by Elastic in this upgrade (target version == base version)
+Then for <field_name> field the diff algorithm should output the current version as the merged one without a conflict
+And <field_name> field should not be returned from the `upgrade/_review` API endpoint
+And <field_name> field should not be shown in the upgrade preview UI
```

Comment:
I think that logic would be pretty straightforward to implement, is there a reason we don't do that first so we wouldn't have to skip the tests initially?

Reply:
@dplumlee Yea of course, if it would be a small and straightforward fix, we could do it in https://github.com/elastic/kibana/pull/184889 along with adding the test coverage 👍